src/meta/src/rpc/metrics.rs

```diff
@@ -923,6 +939,42 @@ impl Default for MetaMetrics {
     }
 }
 
+/// Refresh `system_param_info` metrics by reading current system parameters.
+pub async fn refresh_system_param_info_metrics(
+    system_params_controller: &SystemParamsControllerRef,
+    meta_metrics: Arc<MetaMetrics>,
+) {
+    use risingwave_common::system_param::{
+        BACKUP_STORAGE_URL_KEY, LICENSE_KEY_KEY, STATE_STORE_KEY, system_params_to_kv,
+    };
+
+    fn redact_value_by_name(name: &str, value: &str) -> String {
+        // Replace sensitive values entirely with *** to avoid leaking secrets.
+        if name == LICENSE_KEY_KEY || name == STATE_STORE_KEY || name == BACKUP_STORAGE_URL_KEY {
+            return "***".to_owned();
+        }
+        value.to_owned()
+    }
+
+    let params = system_params_controller.get_pb_params().await;
+    let kvs = match system_params_to_kv(&params) {
+        Ok(kv) => kv,
+        Err(e) => {
+            tracing::warn!("failed to convert system params to kv: {}", e);
+            vec![]
+        }
+    };
+
+    meta_metrics.system_param_info.reset();
+    for (name, value) in kvs {
+        let value = redact_value_by_name(&name, &value);
+        meta_metrics
+            .system_param_info
+            .with_label_values(&[&name, &value])
+            .set(1);
+    }
+}
+
```

Comment:
Please use `SystemParamsRead::get_all` which will handle redaction for you, instead of directly operating on the Protobuf representation.